agent/rpc/peering/service_test.go

```diff
@@ -167,11 +168,13 @@ func TestPeeringService_Initiate(t *testing.T) {
 			req: &pbpeering.InitiateRequest{
 				PeerName:     "peer1-usw1",
 				PeeringToken: validTokenB64,
+				Meta:         map[string]string{"foo": "bar"},
 			},
 			expectResp: &pbpeering.InitiateResponse{},
 			expectPeering: peering.TestPeering(
 				"peer1-usw1",
 				pbpeering.PeeringState_INITIAL,
+				map[string]string{"foo": "bar"},
```

Comment:
I know we have a TODO for the `PeeringWrite` endpoint to potentially be removed: https://github.com/hashicorp/consul/blob/main/proto/pbpeering/peering.proto#L21-L23 But I was wondering if we should add a similar map there to write and then read them back?

Reply:
The way I actually parsed this feedback is that "we should add tests for Peering_Read in the `peering_endpoint` to make sure we are actually storing these in the state store". Please let me know if I got that wrong 🙈 

I added a test for that in the commit [TODO].

Also, we make sure we get the meta data in the API layer as well. So between these two places, I think we can prove that meta data tags get written as expected.